apache/superset
# fix(dashboard): update  cross filter scoping chart id references during dashboard import

during the dashboard import, cross filter scoping is not retained because of not updating chart id references. this commit contains the necessary fix for the same. 

this  pull request provides the fix for the issue 
[#19944 ](https://github.com/apache/superset/issues/19944)
slack discussion: https://apache-superset.slack.com/archives/C015WAZL0KH/p1699248398108929

### SUMMARY
updated the utils.py to update the chart id references for cross filter scoping also. currently chart id reference updates is present for native filters, position and few others, but not for cross filter scoping.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
1. create dashboard with more than 2 charts and apply cross filter scoping.  as shown below. 
2. export the dashboard.
3. if you are importing in the same environment, then delete dashboard and charts. 
4. import the dashboard again. 
5. expectation is cross filter scoping should be retained. 
6. actual is cross filter scoping has been removed and global cross filter scoping changes to all charts. 
dashboard in env1 
![image](https://github.com/apache/superset/assets/80615671/2ff9b3b6-b3ef-4084-859a-dad530568c28)

# BEFORE FIX 
dashboard after importing to env2
![image](https://github.com/apache/superset/assets/80615671/4a039524-730e-4490-aa2f-817fdd3db924)

# AFTER FIX
dashboard after importing to env2
![image](https://github.com/apache/superset/assets/80615671/d5d01b01-a4ad-413a-944c-e09bc97a9240)



### TESTING INSTRUCTIONS
1. create dashboard with cross filter scoping.
2. export the dashboard.
3. delete the dashboard and charts used in dashboards.
4. import the dashboard
5. check the cross filter scoping

### ADDITIONAL INFORMATION

Fixes "#19944" 
- [x] Has associated issue: #19944
